### PR TITLE
workers/social: fix data inconsistencies with account

### PR DESCRIPTION
**Pull request by cihangir**
## Description
Some of the accounts does not have socialapiid so fix that before fixing account issues

## How Has This Been Tested?
manually

## Screenshots (if appropriate):
n/a

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: workers/social/lib/social/render/datafixes.coffee [hunk 1]
```diff
@@ -0,0 +1,70 @@
+module.exports = (client, currentGroup, callback) ->
+
+  async     = require 'async'
+  SocialAccount = require '../../../../../servers/models/socialapi/socialaccount'
```
Comment (gokmen) [inline workers/social/lib/social/render/datafixes.coffee]: `../socialapi/socialaccount`

File: scripts/fix-data-inconsistencies/account.coffee [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            return process.exit 1
+
+          console.log account
+          unless account
+            console.log 'finished'
+            return process.exit 0
+
+          JAccount.one { _id: account._id }, (err, acc) ->
+            if err
+              console.log 'err while getting acc', account._id, err
```
Comment (gokmen) [inline scripts/fix-data-inconsistencies/account.coffee]: need to check `acc` existence here

**Review by gokmen — APPROVED** [review]

File: workers/social/lib/social/render/datafixes.coffee (hunk 1 above)
Comment (cihangir) [inline workers/social/lib/social/render/datafixes.coffee]: blindly copy pasting 🤕

File: workers/social/lib/social/models/account.coffee [hunk 3]
```diff
@@ -301,8 +301,9 @@ module.exports = class JAccount extends jraphical.Module
 
 
   fetchRelativeGroups$: secure (client, callback) ->
-
-    @fetchRelativeGroups callback
+    delegate = client?.connection?.delegate
+    return callback new Error 'malformed request'
```
Comment (gokmen) [inline workers/social/lib/social/models/account.coffee]: `... unless delegate` is missing I believe :)

Comment (gokmen) [inline workers/social/lib/social/models/account.coffee]: actually this must be converted to `permit` usage or at least there must be another check for if requester is this account. this won't work like this. like; `if @equals delegate ...`

Comment (cihangir) [inline workers/social/lib/social/models/account.coffee]: well it seems i forgot to push that change :)

**Comment by cihangir** [comment]
@gokmen please check my latest commit when you have time, merging this and will re-deploy to sandbox & latest

**Pull request merged**